### PR TITLE
Add loading spinner icon overlay when creating font

### PR DESCRIPTION
**Pull request by sakshamarora1**

### CODE REVIEW

File: src/Home.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ function Home(props) {
   const [image, setImage] = useState(["", null]);
   const [font, setFont] = useState("");
   const [fetching, setFetching] = useState(true);
+  const [loading, setLoading] = useState(false);
```
Comment (yashlamba) [inline src/Home.js:13]: We have 2 variables actually tracking the same thing in different ways, Let's remove this redundancy. A better idea would be to make a state variable:
0 -> Not doing anything
1 -> Image uploaded
2 -> Sending image
3 -> Generating font
4 -> Fetching font
5 -> Fetched
6 -> Error

And use them to display the status.

File: src/Home.js [hunk 2]
```diff
@@ -110,46 +112,55 @@ function Home(props) {
...
-                <div>
-                  <h3>Drag 'n' drop your handwritten sample</h3>
-                  <h2>OR</h2>
-                  <center><button type="button" onClick={open}><CloudUploadOutlined />
-                    ‎ ‎ ‎Choose file
-                  </button></center>
-                </div>
-              }
+    <>
+      {!loading ? (
```
Comment (yashlamba) [inline src/Home.js:123]: This would be removed but still mentioning, the condition should be the opposite!?

**Review by yashlamba — CHANGES_REQUESTED** [review]
I was actually confused about how to display this loading but this looks super awesome and neat, I don't think we need an overlay, this looks super nice to me.

File: src/Home.js (hunk 2 above)
Comment (sakshamarora1) [reply src/Home.js:123]: Yes it is for testing right now. Just for checking.

File: src/Home.js [hunk 3]
```diff
@@ -55,18 +65,19 @@ function Home(props) {
...
       else if (response_code === 3) {
-        setFetching(false);
+        setCurrentState(6);
         console.log(message)
       }
       else if (response_code === 0) {
+        setCurrentState(3);
         path = data.path;
         for (let i = 0; i < 10; i++) {
           fetch("http://handwritetest.herokuapp.com/handwrite/status/" + path).then((r) => r.json()).then((status) => {
```
Comment (yashlamba) [inline src/Home.js:83]: We'll need to use async/await here. The requests are not running synchronously.

File: src/Home.js [hunk 4]
```diff
@@ -99,6 +110,7 @@ function Home(props) {
       }
     }).then(() => {
       if (stat === 0) {
```
Comment (yashlamba) [inline src/Home.js:112]: We can get rid of stat now and just use currentState

**Review by yashlamba — CHANGES_REQUESTED** [review]

**Review by yashlamba — APPROVED** [review]

**Pull request merged**